src/components/Chat/InputMessageV2.tsx

```diff
@@ -119,19 +140,200 @@ const InputMessageV2 = ({
+        multiple: true,
+        maxFiles: 20,
+        sortOrder: 'asc',
+        smartAlbums: ['RecentlyAdded', 'UserLibrary']
+      }).then(async (medias) => {
+        const newMedias: any = [];
+        for (const media of medias) {
+          if (media?.mime?.includes('video')) {
+            const thumbnailPath = await getVideoThumbnail(media.path);
+            newMedias.push({
+              type: 'video',
+              path: thumbnailPath,
+              pathVideo: media.path,
+              nameVideo: media.filename,
+              typeVideo: media.mime
+            });
+          } else {
+            const imageCropped = await ImagePicker.openCropper({
+              mediaType: 'photo',
+              path: media.path,
+              width: media.width,
+              height: media.height,
+              cropperChooseText: 'Next',
+              freeStyleCropEnabled: true
+            });
+            newMedias.push({type: 'image', path: imageCropped.path});
+          }
+        }
+
+        handleUploadMedia(newMedias);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your photos, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenCamera = async () => {
+    const {success} = await requestCameraPermission();
+    if (success) {
+      ImagePicker.openCamera({
+        mediaType: 'photo'
+      }).then(async (imageRes) => {
+        const imageCropped = await ImagePicker.openCropper({
+          mediaType: 'photo',
+          path: imageRes.path,
+          width: imageRes.width,
+          height: imageRes.height,
+          cropperChooseText: 'Next',
+          freeStyleCropEnabled: true
+        });
+        handleUploadCamera(imageCropped.path);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your camera, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenFile = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      DocumentPicker.pickSingle({
+        presentationStyle: 'fullScreen'
+      }).then((pickerResult) => {
+        handleFile(pickerResult);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your document library, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
 
   const onChangeInput = (v) => {
     setText(v);
   };
-  const onSelectEmoji = () => {
-    refEmoji.current.close();
+
+  const onOpenGIF = () => {
+    refAttachment.current.close();
+    setTimeout(() => {
+      refGif.current.open();
+    }, 500);
+  };
+
+  const onCloseGIF = () => {
+    refGif.current.close();
+  };
+  const onSelectAttachment = () => {
+    refAttachment.current.open();
   };
 
   const handleSendMessage = () => {
-    onSendButtonClicked(text);
+    onSendButtonClicked(text, []);
+    onSendButtonClicked(text, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This hunk introduces a lot of new functionality related to handling media attachments in chat messages. There are several async functions that handle uploading media, opening the camera, selecting GIFs, etc. These functions seem to be well implemented with proper error handling and state management. However, there are a few points to consider:

- In the `handleUploadMedia` and `handleUploadCamera` functions, the text 'Sent media 🎆🌆🌉' is hardcoded. Consider moving this to a constants file or making it customizable through props.
- The `console.warn('resultUrls', resultUrls);` on line 208 should probably be removed or replaced with a proper logging solution.
- The `getVideoThumbnail` function could potentially return null, but it's not clear how the rest of the code handles this case. Make sure to handle this potential null value appropriately.
- The `onOpenMedia`, `onOpenCamera`, and `onOpenFile` functions request permissions but don't handle the case where the user denies the permission. Consider adding error handling for this case.



<!-- This is an auto-generated comment by OSS CodeRabbit -->